Modules/_functoolsmodule.c

```diff
@@ -38,9 +51,16 @@ partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
         return NULL;
     }
 
+    PyObject *module = PyType_GetModule(type);
```

Comment:
This will get the wrong module for subclasses of `partial`. The *type* is `type(self)`, not `functools.partial`.

But it seems you don't need `_functools_state` in this function?

Reply:
state is used at line 62, no?

Reply:
Right. Sorry for the rushed review.